pkg/genericapiserver/config.go

```diff
@@ -343,48 +352,77 @@ func (c Config) New() (*GenericAPIServer, error) {
 	}
 
 	s.installAPI(&c)
-	s.Handler, s.InsecureHandler = s.buildHandlerChains(&c, http.Handler(s.Mux.BaseMux().(*http.ServeMux)))
+
+	// Wire the handler chain:
+	// - requests from InsecureHandler are marked with skipAuth in the context
```

Comment:
We can discuss whether we really need that downstream: https://github.com/openshift/origin/blob/19ffc82547b383a5811fa7fd9b0990f44647925a/pkg/cmd/server/origin/master.go#L157